shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/config/datasource/DataSourceValidator.java

```diff
@@ -30,23 +31,23 @@
     /**
      * Validate.
      *
-     * @param dataSources data sources.
-     * @return is valid or not
+     * @param dataSourceConfigurations map of data source configuration
+     * @return collection of invalid data sources
      */
-    public boolean validate(final Map<String, DataSourceConfiguration> dataSources) {
```

Comment:
Do you think It is better to rename it as `getInvalidatedDataSources()`? Or maybe we can pass one `DataSourceConfiguration` for validation (my preference).
How do you think?